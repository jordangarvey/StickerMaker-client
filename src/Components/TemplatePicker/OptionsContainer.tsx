import React, { FC } from "react";
import { StyleSheet, View } from "react-native";

import Categories from "./Categories";
import Colours from "./Colours";
import Shapes from "./Shapes";

import ColourPicker from "./OptionViews/ColourPicker";
import ShapePicker from "./OptionViews/ShapePicker";
import TextPicker from "./OptionViews/TextPicker";
import ISelectedOptions from "./ISelectedOptions";


/**
 * Interface for the Options Container props
 */
interface IOptionsContainerProps {
	/** The currently selected options */
	currentOptions: ISelectedOptions;
	/** Callback to be fired then the camera roll button is pressed */
	onCapture(): void;
	/** Callback to be fired whenever the text value changes */
	onChange(text: string): void;
	/** Callback to be fired whenever a option is pressed */
	onPress(category: Categories, option: Colours | Shapes): void;

	/** The currently selected category */
	currentCategory?: Categories;
}

/**
 * Component to render the Options Container
 */
const OptionsContainer: FC<IOptionsContainerProps> = (props) => {
	let options: JSX.Element | null = null;

	switch(props.currentCategory) {
		case Categories.Colour:
			options = <ColourPicker 
				currentColour={props.currentOptions.colour}
				onPress={props.onPress}
			/>;
			break;
		case Categories.Shape:
			options = <ShapePicker 
				currentShape={props.currentOptions.shape}
				onPress={props.onPress}
				selectedColour={props.currentOptions.colour}
			/>;
			break;
		case Categories.Text:
			options = <TextPicker
				currentText={props.currentOptions.text}
				onChange={props.onChange}
			/>;
			break;
	}

	return (
		<View style={styles.optionsContainer}>{options}</View>
	);
};

const styles = StyleSheet.create({
	optionsContainer: {
		height: 160,
		width: "100%"
	}
});

export default OptionsContainer;
