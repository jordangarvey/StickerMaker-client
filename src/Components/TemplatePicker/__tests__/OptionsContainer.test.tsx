import React from "react";

import renderer from "react-test-renderer";

import OptionsContainer from "../OptionsContainer";


describe("The Options Container componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<OptionsContainer/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
