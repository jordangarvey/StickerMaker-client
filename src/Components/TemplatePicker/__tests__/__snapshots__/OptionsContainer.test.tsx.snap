// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Options Container componentâ€™s snapshot renders as expected 1`] = `
<View
  style={
    Object {
      "height": 160,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "column",
        "flexWrap": "wrap",
        "height": "100%",
        "justifyContent": "space-evenly",
      }
    }
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FD5C63",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#00C1B2",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#7289DA",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#EA4C89",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3B5998",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#333333",
            "borderColor": "#333333",
            "borderRadius": 30,
            "borderWidth": 5,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E1306C",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0077b5",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#E0393E",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#00AB6C",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0084FF",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#BD081C",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFC00",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#6441A5",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#1DA1F2",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FF0000",
            "borderRadius": 30,
            "height": 60,
            "marginHorizontal": 15,
            "opacity": 1,
            "width": 60,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
